optuna/cli.py

```diff
@@ -50,6 +48,174 @@ def _check_storage_url(storage_url: Optional[str]) -> str:
+                    header.append(f"{column[0]}_{i}")
+            else:
+                header.append(column[0])
+        for record in records:
+            row = {}
+            for column in columns:
+                if column in record:
+                    value = _format_value(record[column])
+                    if column[1] != "":
+                        row[f"{column[0]}_{column[1]}"] = value
+                    elif any(isinstance(record.get(column), (list, tuple)) for record in records):
+                        for i, v in enumerate(value):
+                            row[f"{column[0]}_{i}"] = v
+                    else:
+                        row[f"{column[0]}"] = value
+            ret.append(row)
+    else:
+        for column in columns:
+            if column[0] not in header:
+                header.append(column[0])
+        for record in records:
+            attrs: Dict[str, Any] = {column_name: {} for column_name in header}
+            for column in columns:
+                if column in record:
+                    value = _format_value(record[column])
+                    if isinstance(column[1], int):
+                        if attrs[column[0]] == {}:
+                            attrs[column[0]] = []
+                        attrs[column[0]] += [None] * max(column[1] + 1 - len(attrs[column[0]]), 0)
+                        attrs[column[0]][column[1]] = value
+                    elif column[1] != "":
+                        attrs[column[0]][column[1]] = value
+                    else:
+                        attrs[column[0]] = value
+            ret.append(attrs)
+
+    return ret, header
+
+
+class ValueType(Enum):
+    NONE = 0
+    NUMERIC = 1
+    STRING = 2
+
+
+class CellValue:
+    def __init__(self, value: Any) -> None:
+        self.value = value
+        if value is None:
+            self.value_type = ValueType.NONE
+        elif isinstance(value, (int, float)):
+            self.value_type = ValueType.NUMERIC
+        else:
+            self.value_type = ValueType.STRING
+
+    def __str__(self) -> str:
+        if isinstance(self.value, datetime.datetime):
+            return self.value.strftime(_DATETIME_FORMAT)
+        else:
+            return str(self.value)
+
+    def width(self) -> int:
+        return len(str(self.value))
+
+    def get_string(self, value_type: ValueType, width: int) -> str:
+        value = str(self.value)
+        if self.value is None:
+            return " " * width
+        elif value_type == ValueType.NUMERIC:
+            return f"{value:>{width}}"
+        else:
+            return f"{value:<{width}}"
+
+
+def _dump_table(records: List[Dict[str, Any]], header: List[str]) -> str:
+    rows = []
+    for record in records:
+        row = []
+        for column_name in header:
+            row.append(CellValue(record.get(column_name)))
+        rows.append(row)
+
+    separator = "+"
+    header_string = "|"
+    rows_string = ["|" for _ in range(len(rows))]
```

Comment:
I think we can simplify this line as follows:
```suggestion
    rows_string = ["|" for _ in rows]
```